src/allencell_ml_segmenter/main/segmenter_layer.py

```diff
@@ -14,9 +14,16 @@ class ShapesLayer(SegmenterLayer):
     data: np.ndarray
 
 
+@dataclass
+class Source(SegmenterLayer):
+    path: Optional[Path] = None
+
+
 @dataclass
 class ImageLayer(SegmenterLayer):
-    path: Optional[Path]
+    path: Optional[Path] = None
+    data: Optional[np.ndarray] = None
+    source: Source = Source(name="sourcename")
```

Comment:
The meaning of 'source' is not clear to me from this file alone. Comments would be helpful. And what is the purpose of the kwarg 'name'?

Reply:
Where do we use the 'source' attribute in the code?